Netcracker/KubeMarine
# [CPDEV-91425] - Remove podman requirement

### Description
To resolve the dependencies conflict between runc and podman we have to install podman and container.io manually before kubemarine installation: https://github.com/Netcracker/KubeMarine/blob/0.19.0/documentation/Installation.md#prerequisites-for-cluster-nodes
But in fact, podman is used only for etcdctl/etcd, and can be replaced using tool `ctr`, that is already installed with containerd.

Fixes # (issue)


### Solution
* podman was excluded from cluster inventory;
* podman configuration was excluded from `prepare.cri.configure` task;
* podman was replaced with `ctr` tool;
* manual steps for rhel were removed from installation guide;
* podman mentions were removed from docs;
* tests were fixed to work without podman;
* patch for etcdctl reinstallation and ctr flags conf file loading were added;

Additionally:
* remove etcd container after `restore` procedure;
* remove ports publishing for ercd container in restore procedure in case of docker cri, because `--network host` already does it;
* `containerdConfig` enrichment was moved to `kubemarine.cri.enrich_inventory` function;


### How to apply
* Can be applied for any nodes;


### Test Cases

**TestCase 1**

Test Configuration: 

- Hardware: 
- OS: rhel 8.3
- Inventory: containerd cri

Steps:

1. fresh `kubemarine install` on empty nodes;

Results:

| Before | After |
| ------ | ------ |
| fails on `prepare.cri.install` task | successful installation |

**TestCase 2**

Test Configuration: 

- Hardware: 
- OS: ubuntu
- Inventory: containerd cri

Steps:

1. run `kubemarine install` on empty nodes;

Results:

| Before | After |
| ------ | ------ |
| podman package will be installed | podman is not installed |

**TestCase 3**

Test Configuration: 

- Hardware: 
- OS: ubuntu
- Inventory: containerd cri

Steps:

1. run `kubemarine install`;
2. Remove `podman`, if it was installed before;
3. run any etcdctl command on nodes, e.g. `sudo etcdctl endpoint status --cluster -w json`;

Results:

| Before | After |
| ------ | ------ |
| etcdctl fails with error `Neither podman nor docker are available to run container, exiting with error...` | etcdctl works |

**TestCase 4**

Test Configuration: 

- Hardware: 
- OS: ubuntu
- Inventory: containerd

Steps:

1. run `kubemarine install`;
2. Remove `podman`, if it was installed before;
3. add test namespace with some objects, e.g. `sudo kubectl create ns test-namespace && sudo kubectl create configmap -n test-namespace test-configmap`;
4. run `kubemarine backup` procedure;
5. remove test namespace, e.g. `sudo kubectl delete ns test-namespace`;
6. run `kubemarine restore` procedure;

Results:

| Before | After |
| ------ | ------ |
| backup fails for `containerd` cri because `docker` and `podman` are not installed | backup-restore works . Removed objects exists in cluster after restoration |

**TestCase 5**

Test Configuration: 

- Hardware: 
- OS: ubuntu
- Inventory: containerd/docker cri

Steps:

1. run `kubemarine install`;
2. run `kubemarine backup` procedure;
3. run `kubemarine restore` procedure;
4. Check created containers using following commands (depends on cri):
```
sudo docker container ls -a
sudo podman ps
sudo ctr container list
```

Results:

| Before | After |
| ------ | ------ |
| etcd container exists after restoration procedure | no containers after restoration |



### Checklist


#### Unit tests
Indicate new or changed unit tests and what they do, if any.
